bin/tx-prover/src/proxy/mod.rs

```diff
@@ -239,6 +254,59 @@ impl LoadBalancer {
         let workers_count = self.num_workers().await;
         Some(create_workers_updated_response(session, workers_count).await)
     }
+
+    /// Create a gRPC client for the given worker address.
+    ///
+    /// It will panic if the worker URI is invalid.
+    async fn create_grpc_client(
+        &self,
+        worker_addr: &str,
+    ) -> Result<HealthClient<Channel>, tonic::transport::Error> {
+        Channel::from_shared(format!("http://{}", worker_addr))
+            .expect("Invalid worker URI")
+            .connect_timeout(self.connection_timeout_secs)
+            .timeout(self.timeout_secs)
+            .connect()
+            .await
+            .map(HealthClient::new)
+    }
+
+    /// Check the health of the workers and returns a list of healthy workers.
+    ///
+    /// Performs a health check on each worker using the gRPC health check protocol. If a worker
+    /// is not healthy, it won't be included in the list of healthy workers.
+    async fn check_workers_health(
+        &self,
+        workers: impl Iterator<Item = &mut Worker>,
+    ) -> Vec<Worker> {
```

Comment:
One thing that we probably want to check in the future (on another PR) is that if a worker does not respond to one health check, it does not necessarily mean it is permanently down (could be related to network transient failures, etc.), so we probably want to have some sort of retries before permanently taking them down.

Reply:
I opened #1009 to address this.